packages/user/src/migrations/Migration20240213092656.ts

```diff
@@ -0,0 +1,54 @@
+import { Migration } from "@mikro-orm/migrations"
+
+export class Migration20240213092656 extends Migration {
+  async up(): Promise<void> {
+    this.addSql(
+      'create table if not exists "invite" ("id" text not null, "email" text not null, "accepted" boolean not null default false, "token" text not null, "expires_at" timestamptz not null, "metadata" jsonb null, "created_at" timestamptz not null default now(), "updated_at" timestamptz not null default now(), "deleted_at" timestamptz null, constraint "invite_pkey" primary key ("id"));'
+    )
+    this.addSql(
+      'alter table "invite" add column if not exists "email" text not null;'
+    )
+
+    this.addSql(`
+    DO $$
+    BEGIN
+      IF EXISTS(SELECT *
+        FROM information_schema.columns
+        WHERE table_name='invite' and column_name='user_email')
+      THEN
+          ALTER TABLE "public"."invite" alter column "user_email" drop not null;
+      END IF;
+    END $$;
+`)
+
+    this.addSql(
+      'create index if not exists "IDX_invite_deleted_at" on "invite" ("deleted_at");'
+    )
+    this.addSql(
+      'CREATE UNIQUE INDEX IF NOT EXISTS "IDX_invite_email" ON "invite" (email) WHERE deleted_at IS NULL;'
+    )
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_invite_token" ON "invite" (token) WHERE deleted_at IS NULL;'
+    )
+
+    this.addSql(
+      'create table if not exists "user" ("id" text not null, "first_name" text null, "last_name" text null, "email" text not null, "avatar_url" text null, "metadata" jsonb null, "created_at" timestamptz not null default now(), "updated_at" timestamptz not null default now(), "deleted_at" timestamptz null, constraint "user_pkey" primary key ("id"));'
+    )
+    this.addSql(
+      'alter table "user" add column if not exists "avatar_url" text null;'
+    )
+
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_user_email" ON "user" (email) WHERE deleted_at IS NULL;'
+    )
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_user_deleted_at" ON "user" (deleted_at);'
+    )
+  }
+
+  async down(): Promise<void> {
+    this.addSql('drop table if exists "invite" cascade;')
+
+    this.addSql('drop table if exists "user" cascade;')
```

Comment:
todo: very dangerous, the down should revert what have been done in the up, here the risk is that if you run the migration of the module and immediately revert, the you lose all the table and data from the core 

cc @olivermrbl @carlos-r-l-rodrigues something to apply to all modules IMO

Reply:
probably copilot could generate the invert of the up

Reply:
well, the up creates the table if it isn't there, then running down wouldn't perform the expected behavior either, a different option would be to create tables like `module_user` instead of `user`? wdyt?